File: CS/CS/Controllers/HomeController.cs
using System;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using CS.Models;

namespace CS.Controllers {
    public partial class HomeController : Controller {
        public ActionResult Index() {
            return View("Index", new BuiltInValidationData());
        }
        [HttpPost]
        public ActionResult Index(FormCollection form) {
            bool isValid = true;
            BuiltInValidationData validationData = new BuiltInValidationData {
                Name = EditorExtension.GetValue<string>("Name", BuiltInValidationHelper.NameValidationSettings, BuiltInValidationHelper.OnNameValidation, ref isValid),
                Age = EditorExtension.GetValue<int?>("Age", BuiltInValidationHelper.AgeValidationSettings, BuiltInValidationHelper.OnAgeValidation, ref isValid),
                Email = EditorExtension.GetValue<string>("Email", BuiltInValidationHelper.EmailValidationSettings, null, ref isValid),
                ArrivalDate = EditorExtension.GetValue<DateTime>("ArrivalDate", BuiltInValidationHelper.ArrivalDateValidationSettings, null, ref isValid)
            };
            if(isValid) {
                return View("ValidationSuccessPartial");
            }
            return View();
        }
    }
}

File: VB/CS/Controllers/HomeController.vb
Imports System
Imports System.Web.Mvc
Imports CS.Models
Imports DevExpress.Web.Mvc

Namespace CS.Controllers
    Partial Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            Return View("Index", New BuiltInValidationData())
        End Function
        <HttpPost> _
        Public Function Index(ByVal form As FormCollection) As ActionResult
            Dim isValid As Boolean = True
            Dim validationData As BuiltInValidationData = New BuiltInValidationData With { _
                .Name = EditorExtension.GetValue(Of String)("Name", BuiltInValidationHelper.NameValidationSettings, AddressOf BuiltInValidationHelper.OnNameValidation, isValid), _
                .Age = EditorExtension.GetValue(Of Integer?)("Age", BuiltInValidationHelper.AgeValidationSettings, AddressOf BuiltInValidationHelper.OnAgeValidation, isValid), _
                .Email = EditorExtension.GetValue(Of String)("Email", BuiltInValidationHelper.EmailValidationSettings, Nothing, isValid), _
                .ArrivalDate = EditorExtension.GetValue(Of Date)("ArrivalDate", BuiltInValidationHelper.ArrivalDateValidationSettings, Nothing, isValid) _
            }
            If isValid Then
                Return View("ValidationSuccessPartial")
            End If
            Return View()
        End Function
    End Class
End Namespace
